### ISSUE TITLE
Angular v2 @Component template errors

### ISSUE THREAD

**Issue by thedailycommute** [eclipse/wildwebdeveloper#523]
Hi.

Every few weeks I install Wild Web Developer, in the hope that it will provide me with a better experience when editing the typescript files associated with our Angular v2 project. And, almost without fail, I end up uninstalling again within 5 minutes!

Why? Two reasons, one minor, one major.

The minor reason is that (perhaps only since I upgrade Eclipse to version 2020-09) there does not seem to be any way to change the font (size and face) used for editing .ts (typescript) files - despite changing _every_ available option in the preferences related to text editors, typescript files continue to display in, what appears to be Segoe UI 9pt. Even the builtin zoom in/zoom out (Window | Editor | Zoom In/Out), which works for all other file types I tried, has no absolutely no effect on .ts files.

The major reason, and the reason I always end up removing Wild Web Developer, has to do with the code validation applied to Angular Component templates.

As noted in the subject line our app still uses Angular v2. It _seems_ to me that Wild Web Developer is validating against a (much) more recent version of Angular because my templates are absolutely riddled with error markers and squiggly red lines. For example the '`<template let-varname="var">`' construct shows an error stating ' "let-" is only supported on ng-template elements. ', and references to other, imported components produces the error:

'component-name' is not a known element:
1. If 'component-name' is an Angular component, then verify that it is part of this module.
2. If 'component-name' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.

There appears to be no way to disable this kind of check, much less configure the version of Angular in use, which makes the Editor completely unusable in practice due to the huge number of erroneous error flags.

These problems are particularly irritating considering that several existing plugins available in the MarketPlace have had their titles changed, for example "Angular Eclipse - Outdated - use Wild Web Developer instead 1.3.0", implying that Wild Web Developer is a dropin replacement for the original.

Sadly this is definitely not the case.

Maybe in a few months I'll try again. But I'm not going to hold my breath...

**Comment by mickaelistria** [comment]
About the font size, it seems like #524 ; can you please verify you're using the proper "Generic Text Editor" and not the TypeScript editor from outdated TypeScript IDE?
About support for Angular v2, Wild Web Developer relying on the official language server from Angular (which is a component of Angular VSCode extension). The latest version, that Wild Web Developer is using, may have dropped support for Angular 2. Maybe try using the tricks in https://github.com/angular/vscode-ng-language-service#versioning to use older version of TypeScript and the angular language service.
In any case, I don't expect anyone to invest in adding specific effort in Wild Web Developer to support older Angular versions.

**Comment by thedailycommute** [comment]
Hi, many thanks for your reply and suggestions.

I'm not bothered about the lack of support for older versions of Angular, what makes the editor unusable is all of the *error* indicators that appear due to the changes in the code validator expecting only newer versions of Angular constructs. What would be more useful would be a way to simply _disable_ Angular (HTML/Template) code validation so that _real_ errors (in the typescript code) can be quickly and easily identified.

I have to confess I was even more disappointed to find, after downloading a clean copy of Eclipse 2020-09, that the Wild Web Developer plugins come integrated into the installation, which makes it impossible to easily edit older projects. Worse still, the original (I know it's deprecated and unsupported) Angular Eclipse plugin 1.3.0 no longer installs completely in a new, clean copy of Eclipse. The old "Typescript" editor is unavailable.

I guess I'm stuck using my (slightly damaged, by the looks of things) old 2020-06 installation (which was originally a 2019-xx instalation that had been successively (and successfully until now), upgraded in-place.

**Comment by mickaelistria** [comment]
You can disable Angular Language Server with Windows > Preferences > Language Servers.
You cann also uninstsall Wild Web Developer totally with Help > Installation Details > Installed Software > Uninstall.
Or you can also move your code to newer Angular to not be annoyed by changes to more modern versions of tools and runtimes.

**Comment by thedailycommute** [comment]
Thanks for pointing me to the Language Servers, I have never seen that setting before.
However, I'm sorry to say it didn't help. After experimenting with practically every combination of enabling/disabling 7 available options:

```
HTML		| HTML Language Server (VSCode)
JavaScript	| JavaScript-TypeScript Language Server
TypeScript	| JavaScript-TypeScript Language Server
JSX		| JavaScript-TypeScript Language Server
TSX		| JavaScript-TypeScript Language Server
TypeScript	| Angular Language Server
HTML		| Angular Language Server

```
I came to the conclusion that the only option that needs to be disabled to stop the template error reports is "TypeScript	| JavaScript-TypeScript Language Server", but that also has the unfortunate side effect of completely disabling TypeScript autocomplete, which defeats the object of using the editor!

I did notice something in the Language Server logs:

> Disabling language service for &lt;project path&gt;/tsconfig.json because it is not an Angular project ('@angular/core/core.d.ts' could not be found). If you believe you are seeing this message in error, please reinstall the packages in your package.json.

The thing is, our project _doesn't have_ a package.json file - it's compiled with sbt, and uses the sbt plugins com.typesafe.play and sbt-typescript, amongst others. I have no idea how (or even if it's possible) to create an equivalent package.json file...

And as for removing Wild Web Developer, that would still leave me without an autocompleting TS editor.

I would _love_ to move the code to a newer version of Angular, but I inherited the project from another programmer who left the company over 2 years ago - the project is built with Angular 2 (v2.0.0-rc5 !!), Play Framework 2.5 (!), Java and a smattering of Scala (my background is C++). A few months ago I did manage to migrate a large part of the code base to Angular 5.2 and Play 2.6, but not 100% because of breaking API changes that I still haven't been able to work around. 

Unfortunately "the powers that be" don't see the need to invest more time and money in upgrading the frameworks when the current version works fine. "If it aint broke, don't fix it". The _app_ may not be broken, but I certainly broke the editor «sigh».
